src/test/regress/expected/isolation_undistribute_table.out

```diff
@@ -251,13 +251,19 @@ step s2-truncate:
 step s1-commit: 
     COMMIT;
 
+s2: WARNING:  relation "public.dist_table" does not exist
```

Comment:
This is unexpected, it is possible that caused by not using `ShouldSyncTableMetadata` (see earlier comment)

Reply:
This is actually expected on a normal pg server.
If you do in single pg node:
```sql
-- connection 1
CREATE TABLE dummy(a int);

BEGIN;
SELECT * FROM dummy;

-- connection 2
TRUCANTE dummy; --blocked

-- connection 1
DROP dummy;
COMMIT;

-- connection 2
-- TRUNCATE unblocks
ERROR: relation dummy does not exist
```

It is the same idea. The relation no longer exists on workers after undistribute, so the error is the same as if a table for which a lock was waiting is no longer there.

Reply:
Ideally, Citus should not try to send `TRUNCATE` commands to the other nodes after the table is undistributed. But, given the `TRUNCATE` is a complicated piece for MX, we have to live with that.

Previous, we were sending `lock_relation_if_exists`, that's why even if the table is not on the worker, we ignored.

I guess, this error is reasonable, similar to what you shared. Would be nice to add some comment to the test spec

Reply:
(This is something to keep in mind when we implement DDL from any node)